src/pages/datasets/[slug].tsx

```diff
@@ -1,29 +1,73 @@
-import type {FunctionComponent} from 'react'
+import {FunctionComponent, useState} from 'react'
+import {useRouter} from 'next/router'
+import {useQuery} from 'react-query'
+
 import {Tag, ButtonGhost} from '@/components/lib'
 import {Tabs, TabList, Tab, TabPanels, TabPanel} from '@reach/tabs'
 import {FileIcon} from '@/components/icons/file'
 
-const Dataset: FunctionComponent<{slug?: string}> = () => {
-  const DatasetData = {
-    title: 'Diabetes Study 01.289.301',
-    description:
-      'This was a double-blind diabetes study done in coordination with UC Santa Barbara between July 1st, 2017 and January 1st, 2019.',
-    tags: ['diabetes', 'california', 'healthcare', 'UCSF', 'beekeeper'],
-    tensors: 2
-  }
+import {IDataset, ITensor} from '@/types/datasets'
+import {fetchDataset} from '../api/datasets'
+import {EditDatasetModal} from '@/components/modals/datasets/edit'
+import {DeleteDatasetModal} from '@/components/modals/datasets/delete'
+
+const Dataset: FunctionComponent = () => {
+  const router = useRouter()
+  const {slug} = router.query
+
+  const [openDeleteDatasetModal, setOpenDeleteDatasetModal] = useState(false)
+  const [openEditDatasetModal, setOpenEditDatasetModal] = useState(false)
+
+  const {isLoading, data: dataset} = useQuery<IDataset, Error>(['dataset', slug], () => fetchDataset({id: slug}))
+
+  if (isLoading || dataset === undefined) return null
 
-  const {title, description, tags} = DatasetData
+  const {id, name, manifest, description, tags, tensors} = dataset
+
+  const handleTensors = (name: string, tensor: ITensor) => (
+    <Tabs key={`tensor-${name}`}>
+      <TabList>
+        <Tab>data</Tab>
+      </TabList>
+      <hr />
+      <TabPanels>
+        <TabPanel>
+          <section className="flex flex-col mt-4 space-y-1">
+            <span>
+              <strong>File</strong>: <FileIcon className="h-4 text-gray-400" /> <span>{name}</span>
+            </span>
+            <span>
+              <strong>Shape</strong>: <span className="text-gray-400">{JSON.stringify(tensor.shape)}</span>
+            </span>
+            <span>
+              <strong>Type</strong>: <span className="text-gray-400">{tensor.dtype}</span>
+            </span>
+            <span>
+              <strong>Schema</strong>:{' '}
+              <span className="text-gray-400">
+                <ul className="list-horizontal">
+                  {tensor.manifest.split(',').map(el => (
+                    <li key={`${el}-element`}>{el}</li>
```

Comment:
This causes problems as `el` is potentially not unique in `tensor.manifest.split(',')`